hastory-server/src/Data/Hastory/Server/Utils.hs

```diff
@@ -0,0 +1,33 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Data.Hastory.Server.Utils where
+
+import Crypto.Hash (Digest, SHA256(..), hashWith)
+import qualified Data.ByteString.Char8 as C
+import Data.Hastory.Types (Entry(..), ServerEntry(..), SyncRequest(..))
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import Network.HostName (HostName)
+
+hashEntry :: Entry -> HostName -> Digest SHA256
+hashEntry Entry {..} host = hashWith SHA256 (unifiedData :: C.ByteString)
+  where
+    unifiedData =
+      mconcat
+        [ T.encodeUtf8 entryText
+        , C.pack $ show entryWorkingDir
+        , C.pack $ show entryDateTime
+        , T.encodeUtf8 entryUser
+        , C.pack host
```

Comment:
I'm being extra annoying about hashing because changing the hashing function later will be a pain.

Reply:
Yup - understood. If we ever do change the hashing function, we'd have to recompute all the hashes, so it will be good to avoid that if we can.